src/ratelimiter.go

```
@@ -189,28 +188,39 @@ func (i *IPRateLimiter) GetLimiter(ip string) (*rate.Limiter, bool) {
 		return rate.NewLimiter(rate.Inf, i.b), true // 白名单中的IP不受限制
 	}
 	
-	// 使用纯IP作为缓存键
+	now := time.Now()
+	
+	// ✅ 双重检查锁定，解决竞态条件
 	i.mu.RLock()
 	entry, exists := i.ips[cleanIP]
```

golangci-lint (1.64.8)

SA4006: this value of `entry` is never used

(staticcheck)